lib/compilers/hlsl.ts

```diff
@@ -71,4 +74,16 @@ export class HLSLCompiler extends BaseCompiler {
     override getIrOutputFilename(inputFilename: string) {
         return this.getOutputFilename(path.dirname(inputFilename), this.outputFilebase).replace('.s', '.dxil');
     }
+
+    override async processAsm(result, filters, options) {
+        if (this.isSpirv(result.asm)) {
+            return this.spirvAsm.processAsm(result.asm, filters);
+        }
+
+        return super.processAsm(result, filters, options);
+    }
+
+    isSpirv(code) {
+        return code.includes('OpMemoryModel') && code.includes('OpEntryPoint');
```

Comment:
minor point; but might be better to do add `code.startsWith('; SPIR-V')`, as you can also trigger this detection by just adding something like this to your code;
```
struct OpMemoryModel {
    float4 test;
};

struct OpEntryPoint {
    float4 test;
};
```

Reply:
I don't think all compilers universally start their output with `; SPIR-V`. However, it's probably good enough for now since DXC is the only HLSL compiler that's currently supporting SPIR-V output.